File: Web_Visualization/src/main/java/com/gatech/project/dao/LanguageSearchCountDAO.java
package com.gatech.project.dao;


import com.gatech.project.domain.LanguageSearchCount;
import com.gatech.project.params.Params;
import com.gatech.project.utils.HBaseUtils;
import org.springframework.stereotype.Component;

import java.util.ArrayList;
import java.util.List;
import java.util.Map;

/**
 * click count DAO
 */
@Component
public class LanguageSearchCountDAO {
    /**
     * Query the data from the HBase
     * @param tableName table name
     * @param day day to fetch
     * @return list of Course Click Count
     */
    public List<LanguageSearchCount> query(String tableName, String day) {
        List<LanguageSearchCount> list = new ArrayList<LanguageSearchCount>();
        // fetch from hbase
        HBaseUtils connect = HBaseUtils.getInstance(Params.QUORUM, Params.PORT);
        Map<String, Long> map = connect.query(Params.COLUMN_FAMILY, Params.COLUMN, tableName, day);
        for(Map.Entry<String, Long> entry : map.entrySet()) {
            LanguageSearchCount model = new LanguageSearchCount();
            if(entry.getKey().length() > 9) {
                model.setName(entry.getKey());
                model.setValue(entry.getValue());
                list.add(model);
            }
        }
        return list;
    }
}

File: Logs_Analysis/src/main/scala/com/gatech/project/dao/LanguageSearchCountDAO.scala
package com.gatech.project.dao

import com.gatech.project.domain.LanguageSearchCount
import com.gatech.project.params.Params
import com.gatech.project.utils.HBaseUtils

import scala.collection.mutable.ListBuffer


/**
 * Language Search DAO Layer
 */
object LanguageSearchCountDAO {

  val tablename = Params.table
  val cf = Params.cf
  val qualifer = Params.column
  val quorum = Params.quorum
  val port = Params.port
  HBaseUtils.setConf(quorum, port)

  /**
   * get the value of language_search
   * @param language_search
   * @return long value
   */
  def count(language_search: String):Long = {
    HBaseUtils.singleGet(this.tablename, language_search, this.cf, this.qualifer)
  }


  /**
   * save to hbase with replacement
   * @param list list of Language Search Count
   */
  def saveOnce(list: ListBuffer[LanguageSearchCount]): Unit = {
    for(ele <- list) {
      HBaseUtils.singlePut(this.tablename, ele.language, this.cf, this.qualifer, ele.search_count)
    }
  }

  /**
   * save to hbase
   * @param list list of Language Search Count
   */
  def save(list: ListBuffer[LanguageSearchCount]): Unit = {
    for(ele <- list) {
      var countNum = ele.search_count + count(ele.language)
      HBaseUtils.singlePut(this.tablename, ele.language, this.cf, this.qualifer, countNum)
    }
  }
}
